### ISSUE TITLE
Components without om/IQuery can incorrectly get next-props passed as nil

### ISSUE THREAD

**Issue by petterik** [omcljs/om#768]
Components without `om/IQuery` can get passed `nil` next-props in `reconcile!`: https://github.com/omcljs/om/blob/master/src/main/om/next.cljc#L2531 inlined code block:
```cjl
2524  (let [computed   (get-computed (props c))
2525        next-raw-props (ui->props env c)
2526        next-props     (om.next/computed next-raw-props computed)]
2527    (when (and (exists? (.-componentWillReceiveProps c))
2528            (iquery? root)
2529             props-change?)
2530      ;; `componentWilReceiveProps` is always called before `shouldComponentUpdate`
2531      (.componentWillReceiveProps c
2532        #js {:omcljs$value (om-props next-props (p/basis-t this))}))
```
I didn't expect this to happen as I wasn't passing `nil` as props to my components.

If I'm understanding this code correctly, it's getting `next-raw-props` from `(ui->props c)` which parses the next props using the component's query and it will always return `nil` for components without query.

This code path makes sense for components with a query, but what should we do about components without a query?
Should those components not reach this code? 
Should components without a query never use props and always use `om/computed` instead? 

Current workaround:
I got around this issue by using `om/computed` instead of props for my components without query. But since computed needs to be attached to a value, and `(ui->props c)` returns `nil` for these components, I pass to my reconciler my own `:ui->props`:
```clj
:ui->props (fn [env c] 
             (if (om/iquery? c) 
               (om/default-ui->props env c)
               {}))
```

**Comment by petterik** [comment]
#766 and #768 are related. Devcards are available here:
https://github.com/petterik/om/tree/om-768-devcard

**Comment by petterik** [comment]
Fixed with 440c527836db747601ec7546ea5f5f349043c9f9
